contracts/Aavegotchi/facets/PolygonXGeistBridgeFacet.sol

```diff
@@ -0,0 +1,135 @@
+                    s.itemTypes[wearableId].totalQuantity += 1;
+                    IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(msg.sender, address(0), address(this), wearableId, 1);
+                } else { // if bridge is vault
+                    LibItems.removeFromOwner(s.itemGeistBridge, wearableId, 1);
+                    IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(msg.sender, s.itemGeistBridge, address(this), wearableId, 1);
+                }
+                LibItems.addToParent(address(this), _tokenId, wearableId, 1);
+                emit LibERC1155.TransferToParent(address(this), _tokenId, wearableId, 1);
+            }
+        }
+    }
+
+    function mint(address _to, uint _tokenId) external onlyGotchiGeistBridge {
+        s.aavegotchis[_tokenId].owner = _to;
+        s.tokenIds.push(uint32(_tokenId));
+        s.ownerTokenIdIndexes[_to][_tokenId] = s.ownerTokenIds[_to].length;
+        s.ownerTokenIds[_to].push(uint32(_tokenId));
+        emit LibERC721.Transfer(address(0), _to, _tokenId);
+    }
+
+    function burn(address _from, uint _tokenId) external onlyGotchiGeistBridge {
+        // burn items before burn gotchi
+        Aavegotchi memory _aavegotchi = s.aavegotchis[_tokenId];
+        for (uint slot; slot < _aavegotchi.equippedWearables.length; slot++) {
+            uint wearableId = _aavegotchi.equippedWearables[slot];
+            if (wearableId != 0) {
+                delete s.aavegotchis[_tokenId].equippedWearables[slot];
+                LibItems.removeFromParent(address(this), _tokenId, wearableId, 1);
+                LibItems.addToOwner(address(0), wearableId, 1);
+                s.itemTypes[wearableId].totalQuantity -= 1;
+
+                IEventHandlerFacet(s.wearableDiamond).emitTransferSingleEvent(msg.sender, address(this), address(0), wearableId, 1);
+                emit LibERC1155.TransferFromParent(address(this), _tokenId, wearableId, 1);
+            }
+        }
+
+        // burn gotchi
+        _aavegotchi.owner = address(0);
+        uint256 index = s.ownerTokenIdIndexes[_from][_tokenId];
+        uint256 lastIndex = s.ownerTokenIds[_from].length - 1;
+        if (index != lastIndex) {
+            uint32 lastTokenId = s.ownerTokenIds[_from][lastIndex];
+            s.ownerTokenIds[_from][index] = lastTokenId;
+            s.ownerTokenIdIndexes[_from][lastTokenId] = index;
+        }
+        s.ownerTokenIds[_from].pop();
+        delete s.ownerTokenIdIndexes[_from][_tokenId];
+
+        // delete token approval if any
+        if (s.approved[_tokenId] != address(0)) {
+            delete s.approved[_tokenId];
+            emit LibERC721.Approval(_from, address(0), _tokenId);
+        }
+
+        emit LibERC721.Transfer(_from, address(0), _tokenId);
+
+        // delete aavegotchi info
+        string memory name = _aavegotchi.name;
+        if (bytes(name).length > 0) {
+            delete s.aavegotchiNamesUsed[LibAavegotchi.validateAndLowerName(name)];
```

Comment:
can people rename gotchis on Geist side?

Reply:
1. disable renaming on polygon
2. migrate all names on Geist
3. enable renaming on Geist
4. ensure names are handled in the metadata transfer